src/node_url.cc

```diff
@@ -47,7 +47,7 @@ enum url_update_action {
   kHref = 9,
 };
 
-void SetArgs(Environment* env, Local<Value> argv[12], const ada::result& url) {
+void SetArgs(Environment* env, Local<Value> argv[13], const ada::result& url) {
```

Comment:
Array arguments decay to pointers so this, in the abstract, doesn't fix anything. You could change it to:
```suggestion
void SetArgs(Environment* env, Local<Value> (*argv)[13], const ada::result& url) {
```
That forces callers to pass a 13-element array by address, i.e.:
```c++
Local<Value> argv[13];
SetArgs(env, &argv, url);
```
You'll need to update all the assignments from `argv[0]` to `(*argv)[0]`.

Reply:
For education purposes: How did the code work before this change?

Reply:
It wrote past the end. Array arguments aren't length-checked; to the compiler `int x[42]` equals `int x[]` equals `int *x`.